tests/testthat/test-check-annotation-values.R

```diff
@@ -424,3 +424,38 @@ test_that("can_coerce() returns FALSE if values aren't coercible", {
   expect_false(can_coerce("a", "logical"))
   expect_false(can_coerce(2.1, "integer"))
 })
+
+test_that("can_coerce() returns TRUE for any capitalization of true/false", {
+  expect_true(can_coerce("true", "logical"))
+  expect_true(can_coerce("True", "logical"))
+  expect_true(can_coerce("TRUE", "logical"))
+  expect_true(can_coerce(TRUE, "logical"))
+  expect_true(can_coerce("false", "logical"))
+  expect_true(can_coerce("False", "logical"))
+  expect_true(can_coerce("FALSE", "logical"))
+  expect_true(can_coerce(FALSE, "logical"))
+})
+
+test_that("can_coerce() returns FALSE for values that aren't true/false", {
+  expect_false(can_coerce("foo", "logical"))
+  expect_false(can_coerce("T", "logical"))
+  expect_false(can_coerce("F", "logical"))
+  expect_false(can_coerce(c("foo", "True"), "logical"))
+  expect_false(can_coerce(c("foo", "True", "FALSE"), "logical"))
+  expect_false(can_coerce(1, "logical"))
+  expect_false(can_coerce(0, "logical"))
+})
+
+test_that("a value can be coerced to its own type", {
+  expect_true(can_coerce(1L, "integer"))
+  expect_true(can_coerce(1, "numeric"))
+  expect_true(can_coerce("foo", "character"))
+  expect_true(can_coerce(TRUE, "logical"))
+  expect_true(can_coerce(factor("foo"), "factor"))
+})
+
+test_that("factors are converted to string before checking coercibility", {
+  can_coerce(factor("TRUE"), "logical")
+  can_coerce(factor("foo"), "character")
+  can_coerce("foo", "character")
+})
```

Comment:
These tests are missing `expect_true()`. Since the last one isn't a factor and is checked elsewhere, it doesn't quite fit in this set, either.

Reply:
Good catch, fixed, thanks!